### PR TITLE
fixes ex_act being called multiple times on the same items, thus destroying it in any explosion strengths

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

see title
also makes items in the epicenter of the explosion fly to a random direction instead of getting flown into the same tile

## Why It's Good For The Game

frag12 instant clothing evaporate is not good

## Changelog
:cl:
fix: fixed ex_act being called until the item is destroyed
/:cl:

### CODE REVIEW

**Comment by jupyterkat** [comment]
probably resolves #3732 , items inside still get damaged though, a bit because zesko decreased it.

File: code/controllers/subsystem/explosion.dm [hunk 1]
```diff
@@ -471,6 +475,7 @@ SUBSYSTEM_DEF(explosions)
 			var/turf/turf_thing = thing
 			turf_thing.explosion_level = max(turf_thing.explosion_level, EXPLODE_LIGHT)
 			turf_thing.ex_act(EXPLODE_LIGHT)
+			lowturf -= turf_thing
 		cost_lowturf = MC_AVERAGE(cost_lowturf, TICK_DELTA_TO_MS(TICK_USAGE_REAL - timer))
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/explosion.dm:479]: Lowturf is an empty list though

File: code/controllers/subsystem/explosion.dm [hunk 2]
```diff
@@ -514,6 +521,7 @@ SUBSYSTEM_DEF(explosions)
 			if(QDELETED(movable_thing))
 				continue
 			movable_thing.ex_act(EXPLODE_DEVASTATE)
+			high_mov_atom -= movable_thing
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/explosion.dm:524]: again high_mov_atom is an empty list

**Review by PowerfulBacon — DISMISSED** [review]
but aint lowturf an empty list?

**Comment by jupyterkat** [comment]
None of the lists you mentioned are empty when I debugged the subsystem

**Comment by PowerfulBacon** [comment]
Because it gets filled between firings.
image

**Comment by PowerfulBacon** [comment]
hm although ex_act might fill it pottentially

**Comment by jupyterkat** [comment]
so the bug is on the modified ex_act procs you requested

**Comment by jupyterkat** [comment]
Because it still removes the items from the list correctly without runtiming for some reason

**Comment by jupyterkat** [comment]
Ex_act calls contents_explosion, which adds the item to the SSexplosions list again

**Comment by francinum** [comment]
Needs investigation. While you're in there can you please just entirely snap zesko's terrible idea to make explosions deal infinite damage to objects immediately in an explosion center?

**Comment by jupyterkat** [comment]
It’s not zesko’s idea, or change honestly. More like a bug that’s brought in through bacon’s or some other maintainer’s review. This addresses that already. 
If you’re talking about the if(src == target) then infinite damage part then that’s more for the bombs deleting itself correctly and not everything on the same tile getting dealt inf damage

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
do things get damaged at all if they are in the contents of a mob now?

**Comment by jupyterkat** [comment]
yes, explosion() calls still add them

**Pull request merged**